jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/mapper/PostMapperDecorator.java

```diff
@@ -30,6 +42,104 @@ public Post postRequestToEntity(PostRequest postRequest, String userName) {
+                        .toList();
+
+                updatePostCommentsRequest.stream()
+                        .filter(postComment -> !newPostComments.contains(postComment))
+                        .forEach((postComment) -> {
+                            postComment.setPost(post);
+                            PostComment mergedPostComment = this.postCommentRepository.save(postComment);
+                            post.getComments().set(post.getComments().indexOf(mergedPostComment), mergedPostComment);
+                        });
+
+                // Add the rows found in the incoming collection,
+                // which cannot be found in the current database snapshot
+                newPostComments.forEach(post::addComment);
+            } else {
+                List<PostComment> comments = new ArrayList<>(post.getComments());
+                for (PostComment comment : comments) {
+                    post.removeComment(comment);
+                }
+            }
+        } else {
+            addPostCommentsToPost(postRequest.comments(), post);
+        }
+        if (!CollectionUtils.isEmpty(post.getTags())) {
+            if (!CollectionUtils.isEmpty(postRequest.tags())) {
+
+                // convertPostCommentsDTO to PostComment Entities
+                List<Tag> updateTagsRequest = this.postMapperDelegate.tagRequestListToTagList(postRequest.tags());
+
+                List<Tag> existingTags =
+                        post.getTags().stream().map(PostTag::getTag).toList();
+
+                // Remove the existing database rows that are no
+                // longer found in the incoming collection (updateTagsRequest)
+                List<Tag> tagsToRemoveList = existingTags.stream()
+                        .filter(tag -> !updateTagsRequest.contains(tag))
+                        .toList();
+                tagsToRemoveList.forEach(post::removeTag);
+
+                List<String> tagNames =
+                        existingTags.stream().map(Tag::getTagName).toList();
+                // Update the existing database rows which can be found
+                // in the incoming collection (updateTagsRequest)
+                List<Tag> newTagsList = updateTagsRequest.stream()
+                        .filter(tag -> !tagNames.contains(tag.getTagName()))
+                        .toList();
+
+                updateTagsRequest.stream()
+                        .filter(tag -> !newTagsList.contains(tag))
+                        .forEach((tag) -> {
+                            tag.setId(existingTags.stream()
+                                    .filter(t -> t.getTagName().equals(tag.getTagName()))
+                                    .findFirst()
+                                    .get()
+                                    .getId());
+                            Tag mergedTag = this.tagRepository.save(tag);
+                            PostTag mergedPostTag = new PostTag(post, mergedTag);
```

Comment:
@coderabbitai, till the code is executed if post.tags() contains 1 value, creating new PostTag object is adding same duplicate value to the list. How to fix this. Please scan the whole module suggest the fix.